extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_2_speech.py

```diff
@@ -0,0 +1,240 @@
+
+        Returns:
+            str: Serialized representation of the prompt and inference settings.
+        """
+        data = copy.deepcopy(data)
+
+        # assume data is completion params for HF text to image task:
+        # https://huggingface.co/docs/diffusers/main/en/api/pipelines/stable_diffusion/text2img#diffusers.StableDiffusionPipeline.__call__
+        # TODO (rossdanlm): Figure out how to check for StableDiffusionXLPipeline
+        prompt_input = data["prompt"]
+
+        # Prompt is handled, remove from data
+        data.pop("prompt", None)
+
+        model_metadata = ai_config.get_model_metadata(data, self.id())
+        prompt = Prompt(
+            name=prompt_name,
+            input=prompt_input,
+            metadata=PromptMetadata(model=model_metadata, parameters=parameters, **completion_params),
+        )
+        return [prompt]
+
+    async def deserialize(
+        self,
+        prompt: Prompt,
+        aiconfig: "AIConfigRuntime",
+        _options,
+        params: Optional[Dict[str, Any]] = {},
+    ) -> Dict[str, Any]:
+        """
+        Defines how to parse a prompt in the .aiconfig for a particular model
+        and constructs the completion params for that model.
+
+        Args:
+            serialized_data (str): Serialized data from the .aiconfig.
+
+        Returns:
+            dict: Model-specific completion parameters.
+        """
+        # Build Completion data
+        model_settings = self.get_model_settings(prompt, aiconfig)
+        [_pipeline_creation_params, unfiltered_completion_params] = refine_pipeline_creation_params(model_settings)
+        completion_data = refine_completion_params(unfiltered_completion_params)
+
+        # Add resolved prompt
+        resolved_prompt = resolve_prompt(prompt, params, aiconfig)
+        completion_data["prompt"] = resolved_prompt
+        return completion_data
+
+    async def run_inference(self, prompt: Prompt, aiconfig: "AIConfigRuntime", options: InferenceOptions, parameters: Dict[str, Any]) -> List[Output]:
+        """
+        Invoked to run a prompt in the .aiconfig. This method should perform
+        the actual model inference based on the provided prompt and inference settings.
+
+        Args:
+            prompt (str): The input prompt.
+            inference_settings (dict): Model-specific inference settings.
+
+        Returns:
+            InferenceResponse: The response from the model.
+        """
+        model_settings = self.get_model_settings(prompt, aiconfig)
+        [pipeline_creation_data, _] = refine_pipeline_creation_params(model_settings)
+        if not pipeline_creation_data.get("requires_safety_checker", True):
+            pipeline_creation_data["safety_checker"] = None
+
+        model_name: str = aiconfig.get_model_name(prompt)
+        if isinstance(model_name, str) and model_name not in self.synthesizers:
+            self.synthesizers[model_name] = pipeline("text-to-speech", model_name)
```

Comment:
Does this take a long time to build sometimes like the text-to-image diffusors? I'm guessing no becuase it's a transformer, not a diffusor but just checking to be sure. It's why I had disclaimer text in https://github.com/lastmile-ai/aiconfig/blob/54de9cc8a1055975b4d28b877bea3dbdc627644e/extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_2_image.py#L274-L282

Reply:
Sometimes it hangs for a long time, but I think that's a bug. When it runs, it seems reasonably fast